File: Programowanie_Obiektowe/PO_JAVA/src/Trawa.java
public class Trawa extends Roslina
{
    public Trawa(int x, int y, Swiat swiat)
    {
        super(Stale.TRAWA_SILA, x, y, swiat);
    }

    @Override
    public void kolizja(Organizm atakujacy)
    {
        swiat.ustawOrganizmNa(atakujacy.getX(), atakujacy.getY(), null);
        atakujacy.setX(x);
        atakujacy.setY(y);
        swiat.ustawOrganizmNa(x, y, atakujacy);

        if (atakujacy.rysowanie() == Stale.OWCA)
        {
            swiat.dodajLog(rysowanie() + " zjedzony przez " + atakujacy.rysowanie() + " (" + atakujacy.getX() + ", " + atakujacy.getY() + ")");
            swiat.usunOrganizm(this);
        }
        else
        {
            swiat.dodajLog(rysowanie() + " pod " + atakujacy.rysowanie() + " (" + atakujacy.getX() + ", " + atakujacy.getY() + ")");
        }
    }

    @Override
    public char rysowanie()
    {
        return Stale.TRAWA;
    }

    @Override
    public String nazwa()
    {
        return Stale.TRAWA_S;
    }

    @Override
    public Organizm stworzPotomka(int x, int y)
    {
        return new Trawa(x, y, swiat);
    }
}

File: Programowanie_Obiektowe/PO_C/PO_C/Trawa.cpp
#include "Trawa.h"

Trawa::Trawa(int x, int y, Swiat& swiat) : Roslina(TRAWA_SILA, x, y, swiat) {}

void Trawa::kolizja(Organizm& atakujacy)
{
    swiat.setLog(swiat.getLog() + string(1, rysowanie()) + " (" + to_string(x) + ", " + to_string(y) + ") pod " + string(1, atakujacy.rysowanie()) + " (" + to_string(atakujacy.getX()) + ", " + to_string(atakujacy.getY()) + ")" + "\n");

    swiat.ustawOrganizmNa(atakujacy.getY(), atakujacy.getX(), nullptr);
    atakujacy.setX(x);
    atakujacy.setY(y);
    swiat.ustawOrganizmNa(atakujacy.getY(), atakujacy.getX(), &atakujacy);

    if (atakujacy.rysowanie() == OWCA)
    {
        swiat.usunOrganizm(this);
    }

    //swiat.usunOrganizm(this);
}

char Trawa::rysowanie() const
{
    return TRAWA;
}

Organizm* Trawa::stworzPotomka(int x, int y) const
{
    return new Trawa(x, y, swiat);
}